### ISSUE TITLE
HDMI induced kernel panic

### ISSUE THREAD

**Issue by RayOfLight1** [tomsom/yoga-linux#60]
Hello, I just bought a 4k 144Hz HDMI2.1 display and if I plug it in my AMD Yoga 7 Gen 7 it works in 4k 30Hz.

It has a setting to enable its EDID to go up to 120Hz, and a further one go to go up to 144Hz. it also affects the YUV and chroma subsampling features.

I've tested other computers, AMD discrete GPUs and the cable, and the cable and display works fine at 4k @ 144Hz. This issue is only happening to me on this laptop, and I've confirmed this issue is happening with linux-6.9.3 and confirmed since 6.6.0 and probably before that.

If I set up the display on such an improved color & refresh rate, as soon as I switch or I plug the cable, my 14ARB7 kernel panics, and I get no output using "journalctl -b -1".

Can you check that your HDMI and post back?

It is an insta-crash on me :(

Thank you!

**Comment by rodude123** [comment]
I'm able to use my laptop connected to a 4k display over HDMI perfectly fine although its only 60hz

**Comment by stuarthayhurst** [comment]
I've tested mine previously on a range of resolutions from 720p -> 4k and it's been fine, but they've all been 60Hz

**Comment by jlo62** [comment]
You shouldn't be able to get over 4k60 (or whatever the max for hdmi 2.0 is), as AMD's hdmi driver shouldn't be able to get up to hdmi 2.1, see [this](https://arstechnica.com/gadgets/2024/02/hdmi-forum-to-amd-no-you-cant-make-an-open-source-hdmi-2-1-driver/)

In my case I can set 3440x1440@100Hz with hdmi from my laptop, but 144Hz with my egpu (with DP, not hdmi, then it's the same)

You should probably open a issue at amd-drm (not for the fps/hdmi 2.1 thingy, but for the kernel panics)\
Can you switch to a TTY and see if you get terminal output for the panic?

And for the 144Hz, either wait till hdmi open-sources the hdmi 2.1 stuff, or go with a thunderbolt cable.

edit: I just realized that the laptop port is only a 2.0 one, so you'll be hit on the head with more then 4k60.\
the TB port has DP 1.4a, which will get you 4k120Hz

**Comment by RayOfLight1** [comment]
Thank you all for your input, after plugging another monitor, system freezes and kernel panics on almost any HDMI plug or unplug event.

I've tried a Kubuntu and Gentoo LiveUSB and they worked perfectly fine.

After some recompiling, I am sure that my custom gentoo-sources kernel, both on 6.6.0 and 6.9.3 have this problem.

I have extracted the kernel config from the Gentoo LiveUSB and I'm now @ 4k @ 120Hz no problem (albeit with chroma subsampling).

I have to do a kompare diff on the original config file and the liveUSB one to see why it behaves like that.

AMDGPU is a module on both cases and I'm a bit lost, but thanks to your input I'm progressing.

Although the kernel should not panic, I'll mark this one as invalid and is not specific for this lenovo unit.
